improve helm chart

fbalicchia opened the pull request:
### CATEGORY

- [x ] Enhancement (new features, refinement)

### SUMMARY
Improve Superset chart

### ADDITIONAL INFORMATION
Hi please review my PR, with this PR I've add possibility to 

* Configure superset_config.py in values using configmap
* Enable/disable Postgres and Redis deploy
* Possibility to add postStart hook for Superset config like create user etc.ect

This is for SupersetNode I will working soon on worker. 

Thanks for your great work

fbalicchia committed:
improve helm chart

fbalicchia committed:
improve helm chart

fbalicchia committed:
add Apache license headers

fbalicchia committed:
add Apache license headers

villebro committed:
fix: suburst chart when secondary metric is defined (#9343)

kristw committed:
refactor: remove settooltip (#9332)

john-bodley committed:
[config] Fixing GET_FEATURE_FLAGS_FUNC example (#9339)

* [config] Fixing GET_FEATURE_FLAGS_FUNC example

* Adding typing for GET_FEATURE_FLAGS_FUNC

* Update config.py

nytai committed:
feat: [SQLLAB] add checkbox to control autocomplete (#9338)

* [SQLLAB] add checkbox to control autocomplete

* autocomplete -> autocompleteEnabled

* fix defaultProps

* fix spec

rusackas committed:
Cal heatmap controls migration (#9355)

* controls migrated

* linting

rusackas committed:
migrating unique EventFlow controls (#9358)

* migrating unique controls

* Lint :sparkles:

rusackas committed:
Treemap controls migration (#9352)

* Treemap

* linting

mistercrunch committed:
feat: [explore] don't save filters inherited from a dashboard (#9340)

* feat: [explore] don't save filters inherited from a dashboard

When navigating to explore from a dashboard context, the current
dashboard filter(s) are passed along to explore so that the context is
kept. So say you're filtering on "country=Romania", in your dashboard
and pivot to explore, that filter is still there and keep on exploring.

Now a common issue is that you'll want to make some tweak to your chart
that are unrelated to the filter, say toggling the legend off for
instance, and then save it. Now you back to your dashboard and even
though you started with an "all countries" dashboard, with a global
filter on country, now that one chart is stuck on "Romania". Typically
you notice this when filtering on something else, say "Italy" and then
that one chart now has two mutually exclusive filters, and show "No data".

Now, the fix is to flag the filter as "extra" (that's the not-so-good internal
name we use for these inherited filters) and make it clear that that
specific filter is special and won't be saved when saving the chart.

* fix build

dpgaspar committed:
[charts] Refactor API using SIP-35 (#9329)

* [charts] Refactor charts API using SIP-35

* [charts] Fix, copy pasta

* [charts] simplify

Commit:
Add global install of webpack and webpack-cli to docker-compose (#9310)

dpgaspar committed:
[dataset] columns and metrics API (nested) (#9268)

* [dataset] columns and metrics API (nested)

* [dataset] tests and validation

* [datasets] Fix, revert list field name to database_name

craig-rueda committed:
Adding requirements-local.txt support (#9357)

* Adding requirements-local.txt support

* Reverting package-lock.json

villebro committed:
Migrate unique Heatmap controls (#9360)

* Migrate Heatmap controls

* Lint

rusackas committed:
fix: don't parseFloat when the *already numeric* value ends in a decimal point (#9351)

john-bodley committed:
[sip-15] Fixing typo in docstring (#9377)

Commit:
[explore view] fix long query issue from Run in SQL LAB Button (#9345)

* [explore view] fix long query issue from Run in SQL LAB Button

* SQL Lab page needs to take the post form data, too

* fix variable names

* updated payload dict, rename hidden form

Co-authored-by: Jesse Yang <jesse.yang@airbnb.com>

Commit:
fix a typo in set prop value (#9383)

rusackas committed:
Migrating unique DirectedForce controls (#9359)

* migrating controls

* Linting :sparkles:

* oops... meant to pull this out in a conflict resolution.

Commit:
Update MANIFEST.in (#9261)

Include superset-frontent/package.json in package because setup.py requires it

nytai committed:
fix: [dashboard] add row padding (#9381)

rusackas committed:
Removing WordCloud controls from CONTRIBUTING.md (#9391)

craig-rueda commented on install/helm/superset/templates/deployment.yaml:
```diff
@@ -42,6 +42,17 @@ spec:
+    {{- end }}
+          lifecycle:
+    {{- if .Values.supersetNode.hooks.postStart }}
+            postStart:
```
Rather than using k8s hooks on the deployment level, you should instead spin a Job up which runs the init script. That way, users can safely tune their replica count up/down without re-triggering the init.

fbalicchia replied:
Thanks, Jobs in Kubernetes will run to completion once they are created. I' will make configurable enable/disable of init job and in the init containers of job image I'll wait for db Do you see any drawback ?

garfieldcoked replied:
If a job fails another instance will be created until the job is completed or max retries reached. I think if the job fails because the database could not be reached and a new instance created after, this could be an alternative to adding logic in the container for waiting on db to be ready.

craig-rueda commented on install/helm/superset/templates/configmap.yaml:
```diff
@@ -24,4 +24,14 @@ metadata:
 data:
-{{ (.Files.Glob "config/*").AsConfig | indent 2 }}
+  {{- $defaultConf := .Files.Get "files/superset_config.py" |  nindent 4 }}
+  superset_config.py: |-
```
This config should really be in a secret. Although the current config has nothing "secret" in it, if a user were to try and use this in a production-like environment, they would likely have secrets in their configs

craig-rueda replied:
Also, pipe the outputs to `tpl` so that end users can use template syntax in their overrides

fbalicchia replied:
Ok

craig-rueda commented on install/helm/superset/values.yaml:
```diff
@@ -54,6 +54,184 @@ resources: {}
+
+    ## (string) Script to execute after the client pod starts.
+    postStart: |-
+      #!/usr/bin/env bash
```
Maybe drop this in a script inside the files folder?

Commit:
[dashboard] handle markdown error (#9350)

* [dashboard] handle markdown error

* localize error message, fix review comments.

john-bodley committed:
[cache] Cleaning up viz/cache logic (#9375)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

rusackas committed:
migrating controls 🎛 (#9386)

rusackas committed:
Migrating unique Table controls (#9388)

rusackas committed:
Migrating unique BoxPlot controls (#9392)

suddjian committed:
upgrade to react-bootstrap v0.33.1 (#9372)

* upgrade to react-bootstrap v0.33.1

* fix test

* simplify proptypes

Co-Authored-By: Evan Rusackas <evan@preset.io>

* simplify more proptypes

Co-Authored-By: Evan Rusackas <evan@preset.io>

Co-authored-by: Evan Rusackas <evan@preset.io>

rusackas committed:
migrating unique controls (#9374)

rusackas committed:
Migrating horizon controls (#9368)

* migrating unique controls

* linting :sparkles:

ktmud committed:
build: use manifest hooks for dev server proxy and fix hot reload for charts (#9333)

* Use manifest hooks for dev server proxy

* Rewrite dashboard/App.jsx to supress Redux error in hot reload

* Update ChartRenderer to allow hot realod in Explore

* Fix hot reload in dashboars as well

* Revert changes to ChartRenderer.jsx

Will submit in another PR.

* Clean up

rusackas committed:
migrating controls (#9373)

dpgaspar committed:
[dataset] New, export API endpoint (#9367)

* [dataset] New, export API endpoint

* Fix, lint

dpgaspar committed:
[charts] New, bulk delete API endpoint (#9387)

* [charts] New, bulk delete API endpoint

* [charts] Fix, typos

* [charts] Fix wrong model name

john-bodley committed:
[sip-15] fix messaging (#9397)

rusackas committed:
Mirgrating unique Partition chart controls (#9385)

* mirgrating controls

* specific import

villebro committed:
feat: add SSL certificate validation for Druid (#9396)

* feat: add SSL certificate feature

* Address comments

* don't mutate extras

* Address comments and add polish

* Add further polish

ChethanUK commented:
> SupersetNode

What is SupersetNode?

john-bodley committed:
[fix] Fixing cache key inconsistencies (#9382)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

fbalicchia commented:
Hi @ChethanUK  with superset node i mean frontend layer not worker. The name come from here
https://github.com/apache/incubator-superset/blob/master/docker-compose.yml#L70 but if have a more appropriate, any suggestion are wellcome

fbalicchia committed:
add initjob conf + secret

fbalicchia committed:
add initjob conf + secret

dpgaspar committed:
[dependency] Fix, Bump FAB to 2.3.1 (#9413)

john-bodley committed:
[mypy] Enforcing typing for charts (#9411)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

fbalicchia committed:
add missing headers

fbalicchia committed:
add missing headers

dpgaspar committed:
[dashboards] Fix, API update slug uniqueness refusing empty string (#9417)

* [dashboards] Fix, API update slug uniqueness refusing empty string

* [dashboards] tests

ktmud committed:
Build: fix hot reload for charts (#9400)

* Fix hot reload for charts

* Override tsc options for plugins

Plugins will sometimes have their own `tsconfig.json`, which will
interfere with the dev server config if configured inappropriately.

This change makes sure `tsc` (and `ts-loader`) always compiles to
`esnext` modules, so we don't get "Uncaught ReferenceError: exports
is not defined" error.

Commit:
[fix] allow force refresh for No Results chart (#9408)

Commit:
[fix] dashboard filter indicator no showing single number value (#9401)

nytai committed:
fix pagination for list views (#9425)

ktmud committed:
feat(visualization): add timeRangeFixed to Big Number w/ Trendlines (#9341)

* feat: add timeRangeFixed and increase font size for Big Number chart

* Update package-lock.json

venter-zhu committed:
[fix]some translation not work better (#9428)

villebro committed:
Add check for SSL certificate and add form validators (#9436)

* Add check for server_cert falsy and add form validators

* Address comments

rusackas committed:
migrating unique controls (#9446)

rusackas committed:
Enforcing linting of LESS (#9356)

* linting some LESS

* enforce linting

* moving font binaries out of stylesheets to fonts folder.

* renaming font specimen sheet directory.

rusackas committed:
Migrating unique bigNumber(total) controls (#9440)

* migrating options controls

* lint :sparkles:

* baby steps...

* shared BigNumber controls file

* capitalization matters

rusackas committed:
Migrating unique NVD3 viz controls (#9445)

* donut control

* labels outside

* linting

* x_log_scale, pie_label_type

* linting

* line_charts, line_charts_2

* marker_labels

* marker_line_labels, marker_lines, markers

* prefix_metric_with_slice_name, range_labels, ranges

* stacked_style

* Revert "stacked_style"

This reverts commit aae2d418b8f1950fbe32248b62e50a12fbf186bd.

* stacked_style take 2...

* Revert "stacked_style take 2..."

This reverts commit 38b7155f50143684c92a67b2a9fa16f65c269d35.

rusackas committed:
Migrating NVD3 Area's stacked_style config (#9454)

* Migrating NVD3 Area's stacked_style config

* test mods

* lint

* nonsense change

* undoing nonsense. Come on GitHub, WORK.

rusackas committed:
Eslint prefer-object-spread (#9466)

* The rule.

* The result

john-bodley committed:
[mypy] Enforcing typing for some modules (#9416)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

ktmud committed:
chore: bump black to 19.10b0 and mypy to 0.770 (#9378)

* Bump black to 19.10b0

* Upgrade mypy to 0.770

* Update how inline type is defined

villebro committed:
release: Add support for ZSH in RELEASING and add tagging instructions (#9451)

* Add support for ZSH in RELEASING and add tagging instructions

* fail if unsupported shell

narendrans committed:
Added dremio (#9472)

dpgaspar committed:
[query] New, readonly API (#9403)

* [query] New, readonly API

* Update tests/queries/api_tests.py

Co-Authored-By: Ville Brofeldt <33317356+villebro@users.noreply.github.com>

john-bodley committed:
[mypy] Enforcing typing for superset.examples (#9469)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

rusackas committed:
Migrating shared DeckGL controls (#9455)

* Missed a couple unique controls!

* new shared module, with filter_nulls export.

* autozoom

* dimension

* nixing unneeded import

* stroke_width

* js_columns

* remaining js_* controls

* cleaning up unused stuff

* legend_format

* exports for prior commit

* legend_position

* line_column

* fill_color_picker

* filled, stroked, stroke_color_picker

* grid_size, extruded

* ... more gridsize

* viewport

* point_radius_fixed, spatial

* multiplier, line_type, line_width

* reverse_long_lat

* lint :sparkles:

narendrans committed:
[Doc] Update installation doc for Dremio (#9464)

* update installation doc for Dremio

Customers fail to install pyodbc properly which on linux systems requires you to install unixodbc-devel package. Update the instructions to make it clear.

* fix new line

dpgaspar committed:
[mypy] Enforcing typing for superset.dashboards (#9418)

* [mypy] Enforcing typing for superset.dashboards

* Make return types equal on all commands

* Make return types equal on all commands

* [dashboard] address comments same return type on commands

* lint

* lint

dpgaspar committed:
[datasets] Add strict type annotation (#9437)

* [datasets] Add strict type annotation

* Fix refresh endpoint

* Improve logic on update

suddjian committed:
Filter owners select by text input (#9337)

* filter owners select by text input

* use rison

* fix backend owners filter logic

* use fullname, not username on owners inputs

* fix some tests

* fixing tests

* deterministic tests

* appease linter

* add back search by username

* more comprehensive filter test

* add clarifying text

* formatting...

elukey committed:
Handle empty dataframes in TableViz (#9480)

TableViz fails to display empty dataframes returning an error like:
"None of [Index(['project', 'count'], dtype='object')] are in the [columns]"

The behavior has been observed while testing 0.36.0rc3 with
Druid datasources.

issue: #9468

dandanhub committed:
[sqllab] Add CUSTOM_TEMPLATE_PROCESSOR config (#9376)

Co-authored-by: Dandan Shi <dshi@dropbox.com>

dpgaspar committed:
[tests] refactor, change datasets and charts to it's own folder (#9487)

villebro committed:
pylint: accept specific 2 character names by default (#9460)

* lint: accept 2 letter names by default

* Address review comments

* Remove e and d from good-names

craig-rueda commented on install/helm/superset/templates/init-job.yaml:
```diff
@@ -0,0 +1,80 @@
+      name: {{ template "superset.name" . }}-init-db
+    spec:
+      initContainers:
+      {{- include "superset.wait-for-postgres" . | nindent 6 }}
```
Rather than doing this via a helper, you should define the `initContainers` block in `values.yaml` and then just use 
```
{{ toYaml | tpl }}
```
here. This way, consumers of this chart can pick their metadata store backend. (some use MySQL, for instance)

craig-rueda commented on install/helm/superset/values.yaml:
```diff
@@ -54,6 +54,132 @@ resources: {}
+    tag: latest
+    pullPolicy: IfNotPresent
+  initscript: |-
+    superset fab create-admin \
```
I think you got things a bit out of order here. Should be:

```
superset db upgrade
superset init
superset fab create-admin ...
```

Also, please add `|| true` to the end of the fab admin command as to make it idempotent. We want this thing to run for each upgrade of Superset so that the DB stays up to date.

craig-rueda reviewed (changes requested):
Few more comments, otherwise looking good

ktmud committed:
Upgrade table chart plugin to 0.12.13 (#9493)

ktmud committed:
feat: change default time range in sql lab explore (#9486)

Change from "100 years go" to "no filter".

100 years ago is basically equivalent to no filter, but has
implications on how x-axis is plotted on certain charts (e.g. Big
Number).

ktmud committed:
fix: add explore control tabOverride at the section level (#9495)

dpgaspar committed:
[query] Migrate api v1 query to new location (#9479)

* [query] Migrate api v1 query to new location

* Improved errors and tests

* [query] nit and revert undesired change

* [query] lint

* [query] revert nan change

* Update superset/queries/api.py

Co-Authored-By: Ville Brofeldt <33317356+villebro@users.noreply.github.com>

* Update tests/queries/api_tests.py

Co-Authored-By: Ville Brofeldt <33317356+villebro@users.noreply.github.com>

* change endpoint location to charts

Co-authored-by: Ville Brofeldt <33317356+villebro@users.noreply.github.com>

dpgaspar committed:
Bump FAB to 2.3.2 (#9491)

dpgaspar committed:
[dashboards] Fix, update dashboard owners not propagating to charts o… (#9484)

* [dashboards] Fix, update dashboard owners not propagating to charts owners

* Add tests

* Fix tests

* better naming

dpgaspar committed:
[dashboards] New, tittle and slug OR filter (#9435)

* [dashboards] New, tittle and slug OR filter

* Update requirements, because of prison bump

* Tests

* Fix tests

* Avoid like filter on empty string value

* merge master brings strict typing to the table

villebro committed:
fix: add lineWidth to Shared_DeckGL.jsx (#9496)

* fix: add lineWidth to Shared_DeckGL.jsx

* linting

fbalicchia committed:
apply changes

fbalicchia committed:
apply changes

fbalicchia committed:
opps again .... license

fbalicchia committed:
opps again .... license

villebro committed:
feat: Add post processing to QueryObject (#9427)

* Add post processing to QueryObject

* Simplify sort signature and require explicit sort order

* Add new operations and unit tests

* linting

* Address comments

* Simplify test method names

* Address comments

* Linting

* remove unnecessary logic

* Apply strict whitelisting to all getattr calls

* Add checking of rolling_type_options and add/improve docs

bkyryliuk committed:
Add visualization flow to the CTA queries (#9370)

* Visualize after running Ctas query (#4)

* Visualize after running Ctas query.

Stub for table viz

Work on the viz

Add JS functionality for Ctas viz

Add test for the table viz flow

Fix test

Add js test for the ctas viz

* Fix tests

* Resolve comments

* Leverate tmpSchema in the query object

* Fix i18n string

Co-authored-by: bogdan kyryliuk <bogdankyryliuk@dropbox.com>

villebro committed:
Bump sqlalchemy and dremio deps (#9503)

dpgaspar committed:
[charts] New, custom filter for name OR description (#9492)

* [charts] New, custom filter for name OR description

* Improve test

octaviancorlade committed:
docker node_modules in its own volume (#9509)

nytai committed:
[Dashboard] new listview filters & emotion infra (#9462)

* [Dashboard] listview filters to match new design

* use title_or_slug filter

* use ?. operator

* move components to components folder

mistercrunch committed:
chore: auto label issues based on the template chosen (#9507)

* chore: auto label issues based on the template chosen

* quoted stuffs

rusackas committed:
control to turn off table bar-chart backgrounds (#9500)

* control to turn off table bar-chart backgrounds

* flipping enable to disable, to support existing charts

* lint :sparkles:

* Turn ON to enable (reversing logic)

* bumping big number chart

* ling :sparkles:

Commit:
Another attempt to fix a viz.py bug (#9522)

* Another attempt to fix a viz.py bug

* Add types to new ddtm attributes

villebro committed:
deprecate groupby controls in query_obj (#9366)

* Deprecate groupby from query_obj

* Fix query_object bug

* Fix histogram

* Remove groupby from legacy druid connector and fix first batch of unit tests

* Deprecate some unnecessary tests + fix a few others

* Address comments

* hide SIP-38 changes behind feature flag

* Break out further SIP-38 related tests

* Reslove test errors

* Add feature flag to QueryContext

* Resolve tests and bad rebase

* Backport recent changes from viz.py and fix broken DeckGL charts

* Fix bad rebase

* backport #9522 and address comments

dpgaspar committed:
[dashboard] Fix, improve test for custom filter (#9521)

dpgaspar committed:
[api] refactor, remove unnecessary code, using command pattern now (#9520)

rusackas committed:
Importing validators module from superset-ui (#9465)

* importing... this is going to break CI, but there it is.

* adding validator package.

* pluralization - maybe it should be plural, but whatevs

* updating validators to match new exports

* removing old validators

* lintage :sparkles:

* validateNumeric -> validateNumber

* using legacyValidateInteger, legacyValidateNumber

* linting :sparkles:

* removing extraneous package lock file

Commit:
Fix typo in viz.py (#9533)

ktmud committed:
[Build] Add Github workflows (#9517)

As an replacement to Travis CI.

craig-rueda committed:
Disabling recording in Cypress tests (#9535)

* Disabling recording in Cypress tests

* Removing --group --ci-build-id

craig-rueda committed:
Run CI on all pushes / PR's (#9538)

nytai committed:
[list views] add work-break css for table layouts (#9537)

dpgaspar committed:
[thumbnails] API and celery task for dashboards and charts (#8947)

axelet committed:
[copy] fix: Row Level Security get_rls_filters func SELECT statement (#9541)

* fix: Row Level Security get_rls_filters func SELECT statement

* More general RowLevelSecurityTests case to avoid improper ids matching

bkyryliuk committed:
Make email parsing more robust (#9523)

* Make email parsing more robust

* Address comment

Co-authored-by: bogdan kyryliuk <bogdankyryliuk@dropbox.com>

mistercrunch committed:
docs: point our README the maintained Docker image (#9536)

Our README now points to a not-so-well-maintain, community-contributed
Docker image. The docker image we link to here is managed and
supported by actual committers under the Preset org.

villebro committed:
Bump copyright notice (#9539)

craig-rueda committed:
Ditching travis config in favor of Github Actions (#9548)

ktmud committed:
Bump @superset-ui/legacy-plugin-chart-table to 0.12.14 (#9549)

dpgaspar committed:
[dashboard] New, add statsd metrics to the API (#9519)

john-bodley committed:
[fix] warm up cache error handling (#9560)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

Commit:
[fix] dashboard filter scope bug (#9550)

ktmud committed:
[Build] Collect frontend code coverage from Cypress tests (#9555)

* build: collect code coverage from Cypress

Collect frontend code coverage reports from Cypress tests and add
proper tagging for all tests.

* Fix bash script lint error from shellcheck

* Revert Cypress to 4.3.0 to see if it fixes a failing test

villebro committed:
chore: Migrate unique FilterBox controls from controls.jsx (#9566)

* Migrate FilterBox controls from controls.jsx

* Linting

villebro committed:
chore: Add OpenAPI docs to /api/v1/chart/data EP (#9556)

* Add OpenAPI docs to /api/v1/chart/data EP

* Add missing fields to QueryObject, fix linting and unit test errors

* Fix unit test errors

* abc

* Fix errors uncovered by schema validation and add unit test for invalid payload

* Add schema for response

* Remove unnecessary pylint disable

dpgaspar committed:
[pypi] Include compiled translations on Pypi pkg (#9567)

villebro committed:
doc: Add changelog for 0.36.0 (#9570)

rusackas committed:
Migrating shared NVD3 controls to new module (#9525)

* proto module

* caught a missed 'freq' unique control

* line_interpolation

* linting

* showLegend

* show_controls

* xAxisLabel

* bottomMargin

* x_ticks_layout

* missed one

* x_axis_format

* yLogScale

* y_axis_bounds

* linting

* nixing yarn lock

* x_axis_showminmax

* xAxisShowminmax control

* richTooltip

* linting, syntax fix

* show_bar_value, bar_stacked

* reduceXticks, yaxislabel

* left_margin, max_bubble_size, y_axis_showminmax

* show_labels

* send_time_range, y_axis_2_format, show_markers, order_bars

* nixing commented imports

* fake controls

* looking up actual controls for comparison.

* adding key to test setup

* controls inventory

* apache junk

* lint :sparkles:

* ignore null controls

* fixing goofed up spread operation for xAxisFormat config

* lint :sparkles:

* fixes for errors caused by <hr> element in filterbox controls

* fixing filter controls for 'druid_time_origin', 'granularity', 'granularity_sqla', 'time_grain_sqla'

* getControlsInventory -> getControlsForVizType

* further renaming of chartControlsInventory - > getControlsForVizType

Co-authored-by: David Aaron Suddjian <aasuddjian@gmail.com>

villebro committed:
fix: Add deprecated fields to QueryObject schema (#9579)

* fix: Add deprecated fields to QueryObject schema

* linting

john-bodley committed:
[mypy] Enforcing typing for superset.migrations (#9583)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

Commit:
Re-enable the AnnotationLayerModelView read API (#9443)

* Re-enable the AnnotationLayerModelView read API

* Fix CI

Commit:
[config] Enable dashboard bootstrap payload reduction by default (#9585)

dpgaspar committed:
[docs] fix, elastic.co elasticsearch product location (#9598)

john-bodley committed:
[fix] Fixing issue with Jinja filter_value (#9582)

Co-authored-by: John Bodley <john.bodley@airbnb.com>

nytai committed:
[Build] moves prettier check to separate script (#9591)

* [Build] moves prettier check to separate script

* rename step: eslint -> lint

Commit:
Add documentation build to Github Actions (#9569)

* Add documentation build to Github Actions

* Update requirements for documentation builds

* Minor optimization - only install requirements for documentation in documentation job

ktmud committed:
Add raw number/integer option to d3 number formats (#9562)

Commit:
Bump superset-ui-connection to 0.12.22 (#9602)

Commit:
[charts] feat: add statsd to charts api (#9571)

* add statsd to charts api

* update test for charts api

* [charts] add statsd asserts wrapper

* [charts] update api test

* removed white space

dpgaspar committed:
[tests] Helper script to run single tests (#9547)

nytai committed:
[charts] adds new filters ui (#9530)

* [charts] adds new filters ui

* move null check to be more visible

* better filter lists and async filter functionality

Commit:
[SQL Lab] Lock result set controls to be always visible (#9587)

craig-rueda commented:
@fbalicchia - can you rebase with master? Looks like GH Actions are stuck. I will merge once status is reporting properly...

fbalicchia committed:
Merge branch 'helm_improve' of github.com:fbalicchia/incubator-superset into helm_improve

fbalicchia commented:
Thanks, @craig-rueda  rebase done. but I think that at moment of writing master is stuck too
